bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java

```diff
@@ -82,7 +98,7 @@ public boolean isManualModeEnabled() {
     @Override
     public void enterManualMode() throws CommunicationApiException {
         try {
-            Http.sendHttpGet(getBaseUrl(), getRequestRequiredOptions() + "&" + CMD_ENABLE_MANUAL_MODE);
+            http.sendHttpGet(getBaseUrl(), getRequestRequiredOptions() + "&" + CMD_ENABLE_MANUAL_MODE);
```

Comment:
Is there a special reason why this binding is not using the shared httpClient? I guess it should be possible to refactor that without too much effort. The idea is that a `HttpClientFactory` is passedas reference to the handler factory, and the shared client https://github.com/openhab/openhab2-addons/blob/1829ddbed0ed2ccfe2ea6dedf2a4a0b74e8979bd/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/AVMFritzHandlerFactory.java#L131 is passed to the handler. The handler uses it like https://github.com/openhab/openhab2-addons/blob/1829ddbed0ed2ccfe2ea6dedf2a4a0b74e8979bd/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/hardware/FritzAhaWebInterface.java#L248-L259

Reply:
Sounds legit, I don't see any reason why the binding needs it's own http client :)

Reply:
The changes for that were a bit bigger, as I made the "pass anything around anything" stuff a bit less annoying. Would like to get your opinion about the approach as well! :)